### PR TITLE
Add missing docstrings

### PR DESCRIPTION
**Pull request by cburgdorf**
### What was wrong?

We have a bunch of types in our docs that are lacking docstrings. I believe having at least *some* docstring per public API is the lowest bar of requirement before we can cut a beta.

### How was it fixed?

Added at least **some** docstring for every single public API.

Imperative present tense is being used for the docstrings.

For methods, it should rhyme with **IT WILL:**
Example:

```python
def get_block:
    """
    Return the block at the tip of the chain
    """
```

For methods, it should rhyme with **IT IS:**
Example:

```python
class SchemaAPI(ABC):
    """
    A class representing a database schema that maps values to lookup keys.
    """
```

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [ ] Clean up commit history

[//]: # (For important changes that should go into the release notes please add a newsfragment file as explained here: https://github.com/ethereum/py-evm/blob/master/newsfragments/README.md)

[//]: # (See: https://py-evm.readthedocs.io/en/latest/contributing.html#pull-requests)
- [x] Add entry to the [release notes](https://github.com/ethereum/py-evm/blob/master/newsfragments/README.md)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: eth/tools/_utils/slow_code_stream.py [hunk 1]
```diff
@@ -65,27 +65,27 @@ def __next__(self) -> int:
             return STOP
 
     def peek(self) -> int:
-        current_pc = self.pc
+        current_pc = self.program_counter
```
Comment (cburgdorf) [inline eth/tools/_utils/slow_code_stream.py]: I have this in a separate commit but thought I could sneak it in here. I personally had to lookup what `pc` stands for again and I remember I already looked it up in the past. Part of this is probably just my poor memory but I also think it aligns better with the rest of the codebase to not use two letter acronyms :)

File: eth/validation.py [hunk 2]
```diff
@@ -174,17 +174,15 @@ def validate_stack_int(value: int) -> None:
...
     )
 
 
 def validate_stack_bytes(value: bytes) -> None:
     if len(value) <= 32:
         return
     raise ValidationError(
-        "Invalid Stack Item: Must be either a length 32 byte "
-        f"string or a 256 bit integer. Got {value!r}"
+        "Invalid Stack Item: Must be either a length 32 byte string. Got {value!r}"
```
Comment (cburgdorf) [inline eth/validation.py]: Another non-doc change that I sneaked in here. I think the text was wrong.

File: eth/vm/code_stream.py [hunk 3]
```diff
@@ -46,31 +46,31 @@ def __getitem__(self, i: int) -> int:
 
     def __iter__(self) -> Iterator[int]:
         # a very performance-sensitive method
-        pc = self.pc
+        pc = self.program_counter
```
Comment (pipermerriam) [inline eth/vm/code_stream.py]: I would extract this to it's own PR.  :+1:

File: eth/abc.py [hunk 4]
```diff
@@ -473,7 +587,7 @@ def add_transaction(self,
     def get_block_transactions(
             self,
             block_header: BlockHeaderAPI,
-            transaction_class: Type[SignedTransactionAPI]) -> Sequence[SignedTransactionAPI]:
+            transaction_class: Type[SignedTransactionAPI]) -> Tuple[SignedTransactionAPI, ...]:
```
Comment (cburgdorf) [inline eth/abc.py]: Another non-doc change sneaked in here. It seems more aligned with our best practices to have the return type be the actual type that is returned (in contrast to parameter types which should accept the broadest type that the method could work with). This also aligns the method with the other methods on that class that do also use `Tuple[Something, ...]` as return type.

File: eth/vm/code_stream.py (hunk 3 above)
Comment (cburgdorf) [reply eth/vm/code_stream.py]: Ooops..too late. I though the approval and rage merged
